### PR TITLE
API-1835: use the "normal" latest revision

### PR DESCRIPTION
**Pull request by deads2k**
brings in https://github.com/openshift/library-go/pull/1801

this brings the revision controller up to current.

### CODE REVIEW

File: pkg/operator/workload/sync_openshift_oauth_apiserver.go [hunk 1]
```diff
@@ -79,16 +78,21 @@ func NewOAuthAPIServerWorkload(
...
+	return c.preconditionFulfilledInternal(operatorSpec, operatorStatus)
 }
 
-func (c *OAuthAPIServerWorkload) preconditionFulfilledInternal(authOperator *operatorv1.Authentication) (bool, error) {
-	argsRaw, err := GetAPIServerArgumentsRaw(authOperator.Spec.OperatorSpec)
+func (c *OAuthAPIServerWorkload) preconditionFulfilledInternal(operatorSpec *operatorv1.OperatorSpec, operatorStatus *operatorv1.OperatorStatus) (bool, error) {
+	if operatorStatus.LatestAvailableRevision == 0 {
+		// this a backstop during the migration from 4.17 whe this information is in .status.oauthAPIServer.latestAvailableRevision
+		return false, fmt.Errorf("waiting for .status.latestAvailableRevision to be available")
+	}
```
Comment (bertinatto) [inline pkg/operator/workload/sync_openshift_oauth_apiserver.go:93]: nice!

**Review by bertinatto — COMMENTED** [review]
/lgtm

File: pkg/operator/workload/sync_openshift_oauth_apiserver.go (hunk 1 above)
Comment (xingxingxia) [reply pkg/operator/workload/sync_openshift_oauth_apiserver.go:93]: @deads2k QE found a Blocker bug OCPBUGS-42932 related  to this.

**Pull request merged**